cassandra/connection.py

```diff
@@ -906,7 +907,7 @@ def _initiate_connection(self, sockaddr):
     #
     # Allow implementations specific to an event loop to add additional behaviours
     def _validate_hostname(self):
-        pass
+        ssl.match_hostname(self._socket.getpeercert(), self.endpoint.address)
```

Comment:
We definitely do NOT want to do this.  ssl.match_hostname() is a deprecated function (and is actually removed in 3.12 IIRC but I don't remember that part for certain) that has been replaced by other functionality.  We leverage that other functionality by setting check_hostname on the ssl_context [here](https://github.com/datastax/python-driver/blob/master/cassandra/connection.py#L870).

PYTHON-1331 has A LOT more detail on this point.